docs/proposals/20220125-ipam-integration.md

```diff
@@ -0,0 +1,410 @@
+---
+title: IP Address Management Integration
+authors:
+  - "@schrej"
+reviewers:
+  - "@fabriziopandini"
+  - "@lubronzhan"
+  - "@maelk"
+  - "@randomvariable"
+creation-date: 2022-01-25
+last-updated: 2022-01-26
+status: provisional
+---
+
+# IP Address Management Integration
+
+## Table of Contents
+
+* [IP Address Management Integration](#ip-address-management-integration)
+   * [Table of Contents](#table-of-contents)
+   * [Glossary](#glossary)
+      * [IPAM Provider](#ipam-provider)
+      * [IP Address Claim](#ip-address-claim)
+      * [IP Address](#ip-address)
+      * [IP Pool](#ip-pool)
+   * [Summary](#summary)
+   * [Motivation](#motivation)
+      * [Goals](#goals)
+      * [Non-Goals/Future Work](#non-goalsfuture-work)
+   * [Proposal](#proposal)
+      * [IPAM API Contract](#ipam-api-contract)
+      * [Pools &amp; IPAM Providers](#pools--ipam-providers)
+      * [Consumption](#consumption)
+      * [User Stories](#user-stories)
+         * [Story 1](#story-1)
+         * [Story 2](#story-2)
+         * [Story 3](#story-3)
+         * [Story 4](#story-4)
+      * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Security Model](#security-model)
+      * [Risks and Mitigations](#risks-and-mitigations)
+   * [Alternatives](#alternatives)
+   * [Upgrade Strategy](#upgrade-strategy)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+### IPAM Provider
+
+A component that provides IP Address Management to Cluster API resources. Examples include in-cluster management of IPs through custom resources, or integration with external systems like Netbox or Infoblox.
+
+### IPAddressClaim
+
+An IPAddressClaim is a custom resource that can be used by infrastructure providers to allocate IP addresses from IP Pools that are managed by IPAM providers.
+
+### IPAddress
+
+An IPAddress is a custom resource that gets created by an IPAM provider to fulfil an IPAddressClaim, which then gets consumed by the infrastructure provider that created the claim.
+
+### IP Pool
```

Comment:
IPPool?

Reply:
+1

Reply:
Intentional, see [here](https://github.com/kubernetes-sigs/cluster-api/pull/6000#discussion_r794425270). There won't be an `IPPool` resource that's part of CAPI, so I wanted it to be clear that it isn't a resource but a category of resources.